File: frontend/src/services/appointmentService.js
import apiClient from './apiClient';

const appointmentService = {
  async book(payload) {
    const { data } = await apiClient.post('/appointments', payload);
    return data.data;
  },
  async getById(id) {
    const { data } = await apiClient.get(`/appointments/${id}`);
    return data.data;
  },
  async getByPatient(patientId, params = {}) {
    const { data } = await apiClient.get(`/appointments/patient/${patientId}`, { params });
    return data.data;
  },
  async getByDoctor(doctorId, params = {}) {
    const { data } = await apiClient.get(`/appointments/doctor/${doctorId}`, { params });
    return data.data;
  },
  async getDoctorScheduleForDate(doctorId, date) {
    const { data } = await apiClient.get(`/appointments/doctor/${doctorId}/schedule`, { params: { date } });
    return data.data;
  },
  async updateStatus(id, status) {
    const { data } = await apiClient.patch(`/appointments/${id}/status`, { status });
    return data.data;
  },
  async getTodaysAppointments() {
    const { data } = await apiClient.get('/appointments/today');
    return data.data;
  },
  async cancel(id, reason) {
    const { data } = await apiClient.patch(`/appointments/${id}/cancel`, null, { params: { reason } });
    return data.data;
  },
};

export default appointmentService;

File: backend/src/main/java/com/hospital/hms/service/AppointmentService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.request.AppointmentRequest;
import com.hospital.hms.dto.request.AppointmentStatusUpdateRequest;
import com.hospital.hms.dto.response.AppointmentResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;
import java.util.List;

public interface AppointmentService {

    AppointmentResponse book(AppointmentRequest request, Long bookedByUserId);

    AppointmentResponse updateStatus(Long id, AppointmentStatusUpdateRequest request);

    AppointmentResponse cancel(Long id, String reason);

    AppointmentResponse getById(Long id);

    Page<AppointmentResponse> getByPatient(Long patientId, Pageable pageable);

    Page<AppointmentResponse> getByDoctor(Long doctorId, Pageable pageable);

    List<AppointmentResponse> getDoctorScheduleForDate(Long doctorId, LocalDate date);

    List<AppointmentResponse> getTodaysAppointments();
}
